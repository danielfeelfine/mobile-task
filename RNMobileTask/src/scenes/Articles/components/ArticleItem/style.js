import { StyleSheet } from 'react-native'
import { colors } from '@styles/base'

const styles = StyleSheet.create({
  item: {
    flex: 1,
    flexDirection: 'row',
    paddingVertical: 8,
    paddingHorizontal: 8,
    backgroundColor: colors.Light,
  },
  imageContent: {
    marginRight: 10
  },
  image: {
    width: 88,
    height: 84
  },
  textContent: {
    flex: 1,
    height: 84
  },
  name: {
    marginBottom: 2,
    fontSize: 14,
    fontWeight: '800',
    color: colors.Dark,
  },
  description: {
    marginVertical: 2,
    fontSize: 12,
    fontWeight: '300',
    color: colors.Dark,
  },
  url: {
    marginVertical: 2,
    fontSize: 10,
    fontWeight: '300',
    color: colors.Gray,
  }
})

export default styles
