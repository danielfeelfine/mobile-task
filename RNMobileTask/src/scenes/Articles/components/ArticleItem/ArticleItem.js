// @flow
import React from 'react'
import { Text, View, Image, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import { normalizeImageURL } from '@helpers/normalize'
import images from '@assets/images'
import style from './style'


type Props = {
  title: string,
  author: string,
  description: string,
  sourceURL: string,
  imageURL: string,
  onPress: Function
}

const ArticleItem = (props: Props) => {
  return (
    <TouchableOpacity style={style.item} onPress={props.onPress}>
      <View style={style.imageContent}>
        <Image
          style={style.image}
          resizeMethod="resize"
          source={(!props.imageURL ? images.NotAvailableImage : { uri: normalizeImageURL(props.sourceURL, props.imageURL) })}
        />
      </View>

      <View style={style.textContent}>
        <Text style={style.name} numberOfLines={2}>{props.title}</Text>
        <Text style={style.url}>{props.author}</Text>
        <Text style={style.description} numberOfLines={2}>{props.description}</Text>
      </View>
    </TouchableOpacity>
  )
}

ArticleItem.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  description: PropTypes.string.isRequired,
  sourceURL: PropTypes.string.isRequired,
  imageURL: PropTypes.string,
  onPress: PropTypes.func
}

export default ArticleItem
