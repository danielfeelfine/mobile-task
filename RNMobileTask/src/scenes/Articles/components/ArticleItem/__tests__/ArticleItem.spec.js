import React from 'react'
import { shallow } from 'enzyme'
import ArticleItem from './../ArticleItem'


describe('ArticleItem component', () => {

  it('renders correctly', () => {
    const props = {
      title: 'Item name',
      author: 'Item name',
      description: 'Item description',
      sourceURL: 'https://website.news/',
      imageURL: 'https://website.news/static/whatever.jpg'
    }

    const wrapper = shallow(
      <ArticleItem
        title={props.title}
        author={props.author}
        description={props.description}
        sourceURL={props.sourceURL}
        imageURL={props.imageURL}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

})
