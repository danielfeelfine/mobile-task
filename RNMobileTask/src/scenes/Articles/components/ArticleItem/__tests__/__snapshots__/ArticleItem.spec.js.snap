// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleItem
    author="Item name"
    description="Item description"
    imageURL="https://website.news/static/whatever.jpg"
    sourceURL="https://website.news/"
    title="Item name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": Array [
        <Component
          style={
            Object {
              "marginRight": 10,
            }
          }
        >
          <Image
            resizeMethod="resize"
            source={
              Object {
                "uri": "https://website.news/static/whatever.jpg",
              }
            }
            style={
              Object {
                "height": 84,
                "width": 88,
              }
            }
          />
        </Component>,
        <Component
          style={
            Object {
              "flex": 1,
              "height": 84,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Object {
                "color": "#333333",
                "fontSize": 14,
                "fontWeight": "800",
                "marginBottom": 2,
              }
            }
          >
            Item name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
                "fontSize": 10,
                "fontWeight": "300",
                "marginVertical": 2,
              }
            }
          >
            Item name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "300",
                "marginVertical": 2,
              }
            }
          >
            Item description
          </Text>
        </Component>,
      ],
      "onPress": undefined,
      "style": Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 8,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            resizeMethod="resize"
            source={
              Object {
                "uri": "https://website.news/static/whatever.jpg",
              }
            }
            style={
              Object {
                "height": 84,
                "width": 88,
              }
            }
          />,
          "style": Object {
            "marginRight": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMethod": "resize",
            "source": Object {
              "uri": "https://website.news/static/whatever.jpg",
            },
            "style": Object {
              "height": 84,
              "width": 88,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Object {
                  "color": "#333333",
                  "fontSize": 14,
                  "fontWeight": "800",
                  "marginBottom": 2,
                }
              }
            >
              Item name
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontSize": 10,
                  "fontWeight": "300",
                  "marginVertical": 2,
                }
              }
            >
              Item name
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Object {
                  "color": "#333333",
                  "fontSize": 12,
                  "fontWeight": "300",
                  "marginVertical": 2,
                }
              }
            >
              Item description
            </Text>,
          ],
          "style": Object {
            "flex": 1,
            "height": 84,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Item name",
              "ellipsizeMode": "tail",
              "numberOfLines": 2,
              "style": Object {
                "color": "#333333",
                "fontSize": 14,
                "fontWeight": "800",
                "marginBottom": 2,
              },
            },
            "ref": null,
            "rendered": "Item name",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Item name",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#999999",
                "fontSize": 10,
                "fontWeight": "300",
                "marginVertical": 2,
              },
            },
            "ref": null,
            "rendered": "Item name",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Item description",
              "ellipsizeMode": "tail",
              "numberOfLines": 2,
              "style": Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "300",
                "marginVertical": 2,
              },
            },
            "ref": null,
            "rendered": "Item description",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Component
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <Image
              resizeMethod="resize"
              source={
                Object {
                  "uri": "https://website.news/static/whatever.jpg",
                }
              }
              style={
                Object {
                  "height": 84,
                  "width": 88,
                }
              }
            />
          </Component>,
          <Component
            style={
              Object {
                "flex": 1,
                "height": 84,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Object {
                  "color": "#333333",
                  "fontSize": 14,
                  "fontWeight": "800",
                  "marginBottom": 2,
                }
              }
            >
              Item name
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontSize": 10,
                  "fontWeight": "300",
                  "marginVertical": 2,
                }
              }
            >
              Item name
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Object {
                  "color": "#333333",
                  "fontSize": 12,
                  "fontWeight": "300",
                  "marginVertical": 2,
                }
              }
            >
              Item description
            </Text>
          </Component>,
        ],
        "onPress": undefined,
        "style": Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              resizeMethod="resize"
              source={
                Object {
                  "uri": "https://website.news/static/whatever.jpg",
                }
              }
              style={
                Object {
                  "height": 84,
                  "width": 88,
                }
              }
            />,
            "style": Object {
              "marginRight": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMethod": "resize",
              "source": Object {
                "uri": "https://website.news/static/whatever.jpg",
              },
              "style": Object {
                "height": 84,
                "width": 88,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#333333",
                    "fontSize": 14,
                    "fontWeight": "800",
                    "marginBottom": 2,
                  }
                }
              >
                Item name
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#999999",
                    "fontSize": 10,
                    "fontWeight": "300",
                    "marginVertical": 2,
                  }
                }
              >
                Item name
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={2}
                style={
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                    "fontWeight": "300",
                    "marginVertical": 2,
                  }
                }
              >
                Item description
              </Text>,
            ],
            "style": Object {
              "flex": 1,
              "height": 84,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Item name",
                "ellipsizeMode": "tail",
                "numberOfLines": 2,
                "style": Object {
                  "color": "#333333",
                  "fontSize": 14,
                  "fontWeight": "800",
                  "marginBottom": 2,
                },
              },
              "ref": null,
              "rendered": "Item name",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Item name",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#999999",
                  "fontSize": 10,
                  "fontWeight": "300",
                  "marginVertical": 2,
                },
              },
              "ref": null,
              "rendered": "Item name",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Item description",
                "ellipsizeMode": "tail",
                "numberOfLines": 2,
                "style": Object {
                  "color": "#333333",
                  "fontSize": 12,
                  "fontWeight": "300",
                  "marginVertical": 2,
                },
              },
              "ref": null,
              "rendered": "Item description",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
